src/main/scala/com/dataintuitive/viash/functionality/Functionality.scala

```diff
@@ -259,11 +259,15 @@ case class Functionality(
   require(name.matches("^[A-Za-z][A-Za-z0-9_]*$"), message = "functionality name must begin with a letter and consist only of alphanumeric characters or underscores.")
 
   // check arguments
+  val allNames = allArguments.map(a => a.name) ::: allArguments.flatMap(a => a.alternatives)
+  val allNamesCounted = allNames.groupBy(identity).map(a => (a._1, a._2.length))
```

Comment:
Do you intend for these arguments to be exposed? If not, you can surround them in a code block (similar to other temporary values used in other argument value assertions).